File: java/src/org/gessinger/gepard/xmp/EmitterWithStatusInfo.java
package org.gessinger.gepard.xmp ;
import org.gessinger.gepard.* ;
import java.util.Map ;
public class EmitterWithStatusInfo
{
  static public void main ( String[] args )
  {
    Util.argsToProperties ( args ) ;
    try
    {
      EmitterWithStatusInfo j = new EmitterWithStatusInfo() ;
    }
    catch ( Exception exc )
    {
      exc.printStackTrace() ;
    }
  }
  EmitterWithStatusInfo()
  throws Exception
  {
    Client client = Client.getInstance() ;
    String name = Util.getProperty ( "name", "ALARM" ) ;

    Event e = new Event ( name ) ;
    client.emit ( name, new StatusCallback()
    {
      public void status ( Event e )
      {
        System.out.println ( "status=" + e.getStatusName() ) ;
        System.out.println ( e.getStatusReason() ) ;
        e.getClient().close() ;
      }
    } ) ;
  }
}

File: python/xmp/EmitterWithStatusInfo.py
#!/usr/bin/env python

import os, sys
sys.path.insert ( 0, os.path.dirname(os.path.abspath(__file__) ) + "/../" )

from gepard import Event, Client

# ==========================================================================

client = Client.getInstance()

def on_error ( err, info ):
	print ( err )

def status ( event ):
	print ( "status=" + event.getStatusName() )
	print ( event.getStatusReason() )
	event.getClient().close()

client.onError ( on_error )

client.emit ( "ALARM", status=status )
